libs/api/protocol/src/lib/core/query/get-protocol-entries.query.ts

```diff
@@ -0,0 +1,70 @@
+import { Inject } from '@nestjs/common';
+import { IQueryHandler, QueryHandler } from '@nestjs/cqrs';
+
+import { Page } from '../entity/page.entity';
+import { ProtocolEntryBase } from '../entity/protocol-entries/protocol-entry-base.entity';
+import {
+	PROTOCOL_ENTRY_REPOSITORY,
+	ProtocolEntryRepository,
+} from '../repository/protocol-entry.repository';
+
+export class GetProtocolEntriesQuery {
+	constructor(
+		public readonly organizationId: string,
+		public readonly count: number,
+		public readonly sort: 'asc' | 'desc',
+		public readonly startingFrom?: Date,
+	) {}
+}
+
+@QueryHandler(GetProtocolEntriesQuery)
+export class GetProtocolEntriesHandler
+	implements IQueryHandler<GetProtocolEntriesQuery>
+{
+	constructor(
+		@Inject(PROTOCOL_ENTRY_REPOSITORY)
+		private readonly repository: ProtocolEntryRepository,
+	) {}
+
+	async execute({
+		organizationId,
+		count,
+		sort,
+		startingFrom,
+	}: GetProtocolEntriesQuery): Promise<Page<ProtocolEntryBase>> {
+		const protocolEntries = await this.repository.getProtocolEntries(
+			organizationId,
+			count,
+			sort,
+			startingFrom,
+		);
+
+		if (sort === 'asc') {
```

Comment:
I moved the `reverse` to the repository level as this should not be a concern of the handler. 

Unfortunately, we can't quite move it all the way to the database as there is no simple "getLastN". There are similiar capabilities with MongoDB pipelines, however, this would introduce a lot of overhead codewise and inflict additional costs upon the database just to spare the relatively cheap `reverse` call. That's why I would like to keep it that way.